pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -31,7 +31,7 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 )
 
-var supportedScoringStrategyTypes = sets.NewString(
+var supportedScoringStrategyTypes = sets.New(
```

Comment:
Another thought (sorry, didn't occur to me sooner): `supportedScoringStrategyTypes` doesn't need to be a set of strings, does it?

Removing all of the casting to `string` was necessary before because we only had `sets.String`. Now can create sets of arbitrary types. So perhaps try removing the casting here and in those places which use `supportedScoringStrategyTypes`?

May also apply elsewhere.

Reply:
didnt get you

Reply:
Just a thought wouldnt it be better if what you are saying is covered in different PR. As this Pr is already pretty huge.

Reply:
I think the problem is that we use the set like this `sets.List(supportedScoringStrategyTypes)` to pass it to `field.Unsupported`. Maybe we need a generic method that can cast lists into other types. But we don't have it today.

Reply:
That `field.Unsupported` needs a `[]string` for the valid values indeed ruins my cunning plan. Let's leave it as-is, but perhaps add a comment to `supportedScoringStrategyTypes` about "supportedScoringStrategyTypes has to be a set of strings for use with field.Unsupported`? Then the next person looking at it will know that trying to simplify it will not work and not waste time on suggesting and/or trying it.

Reply:
or somebody could introduce a `list` package that knows how to do type conversions :)

```golang
package list

func Cast[A, B](in []A) ([]B) {
```